### PR TITLE
feat(clerk-js): Add dev mode notice to components

### PR DESCRIPTION
**Pull request by desiprisg**
## Description

Add a development mode notice to our components to avoid going to productions with dev keys by accident.

CleanShot 2024-07-25 at 15 30 12@2x

CleanShot 2024-07-12 at 03 22 47@2x

CleanShot 2024-07-25 at 15 31 12@2x

CleanShot 2024-07-31 at 13 08 00@2x

CleanShot 2024-07-31 at 13 08 33@2x

## Checklist

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

**Comment by desiprisg** [comment]
!preview

**Comment by clerk-cookie** [comment]
Hey @desiprisg, your preview is currently deploying and should be available shortly.

| Status | Preview | Updated (UTC) |
| :-- | :-- | :-- |
| :cook: _Deploying..._ | [Inspect](https://javascript-preview-idl0f5wqv.clerkstage.dev) | Jun 04, 2024 10:05 AM |

**Comment by desiprisg** [comment]
!preview

File: packages/clerk-js/src/ui/elements/DevModeNotice.tsx
Comment (PixelJanitor) [inline packages/clerk-js/src/ui/elements/DevModeNotice.tsx:18]: You can remove this width, height, top, and left and replace with just `inset: 0` 👍

File: packages/clerk-js/src/ui/elements/DevModeNotice.tsx
Comment (PixelJanitor) [inline packages/clerk-js/src/ui/elements/DevModeNotice.tsx:24]: rgb is fine here but for readability would just reccomend `black`.

File: packages/clerk-js/src/ui/elements/Card/CardContent.tsx
Comment (PixelJanitor) [inline packages/clerk-js/src/ui/elements/Card/CardContent.tsx:90]: Seems like we can just `left: 0, right: 0, textAlign: 'center'` here instead of the shift left and pull back with transform?

**Comment by anagstef** [comment]
@desiprisg can we have a screenshot of how this looks like?

**Comment by desiprisg** [comment]
> @desiprisg can we have a screenshot of how this looks like?

Adding these! Feel free to check the preview deployment as well!

File: packages/clerk-js/src/ui/elements/Navbar.tsx
Comment (anagstef) [inline packages/clerk-js/src/ui/elements/Navbar.tsx:177]: ❓ Wouldn't the removal of the padding cause issues when the devmode notice is not shown?

File: packages/clerk-js/src/ui/elements/PopoverCard.tsx
Comment (anagstef) [inline packages/clerk-js/src/ui/elements/PopoverCard.tsx:95]: Same as above

**Review by anagstef — APPROVED** [review]

**Comment by desiprisg** [comment]
> Hey @desiprisg , whats the status of this one?

We were waiting for user button / org switcher designs before merging this one.

**Comment by octoper** [comment]
!preview

**Comment by nikosdouvlis** [comment]
@octoper hey :) is this good to go? if not, do you need help with anything at this point?

**Comment by octoper** [comment]
There were no designs for `<OrgSwitcher/>` and `<UserButton/>` until yesterday, I will take a look at this today

**Comment by octoper** [comment]
!preview

**Comment by octoper** [comment]
!preview

**Comment by kylemac** [comment]
@octoper do we offer the ability for a developer to turn this off in Dev?

**Comment by octoper** [comment]
!preview

**Comment by octoper** [comment]
!preview

**Comment by octoper** [comment]
!preview

**Comment by octoper** [comment]
!preview

**Comment by anagstef** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @anagstef - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/astro | 0.0.3-snapshot.v2af7822 |
| @clerk/backend | 1.3.2-snapshot.v2af7822 |
| @clerk/chrome-extension | 1.1.4-snapshot.v2af7822 |
| @clerk/clerk-js | 5.9.0-snapshot.v2af7822 |
| @clerk/elements | 0.10.3-snapshot.v2af7822 |
| @clerk/clerk-expo | 1.2.7-snapshot.v2af7822 |
| @clerk/express | 0.0.17-snapshot.v2af7822 |
| @clerk/fastify | 1.0.19-snapshot.v2af7822 |
| gatsby-plugin-clerk | 5.0.0-beta.45 |
| @clerk/localizations | 2.4.8-snapshot.v2af7822 |
| @clerk/nextjs | 5.2.4-snapshot.v2af7822 |
| @clerk/clerk-react | 5.2.8-snapshot.v2af7822 |
| @clerk/remix | 4.2.3-snapshot.v2af7822 |
| @clerk/clerk-sdk-node | 5.0.16-snapshot.v2af7822 |
| @clerk/shared | 2.3.3-snapshot.v2af7822 |
| @clerk/tanstack-start | 0.1.7-snapshot.v2af7822 |
| @clerk/testing | 1.1.11-snapshot.v2af7822 |
| @clerk/themes | 2.1.12-snapshot.v2af7822 |
| @clerk/types | 4.8.0-snapshot.v2af7822 |

Tip: Use the snippet copy button below to quickly install the required packages.
`@clerk/astro`
```sh
npm i @clerk/astro@0.0.3-snapshot.v2af7822 --save-exact
```
`@clerk/backend`
```sh
npm i @clerk/backend@1.3.2-snapshot.v2af7822 --save-exact
```
`@clerk/chrome-extension`
```sh
npm i @clerk/chrome-extension@1.1.4-snapshot.v2af7822 --save-exact
```
`@clerk/clerk-js`
```sh
npm i @clerk/clerk-js@5.9.0-snapshot.v2af7822 --save-exact
```
`@clerk/elements`
```sh
npm i @clerk/elements@0.10.3-snapshot.v2af7822 --save-exact
```
`@clerk/clerk-expo`
```sh
npm i @clerk/clerk-expo@1.2.7-snapshot.v2af7822 --save-exact
```
`@clerk/express`
```sh
npm i @clerk/express@0.0.17-snapshot.v2af7822 --save-exact
```
`@clerk/fastify`
```sh
npm i @clerk/fastify@1.0.19-snapshot.v2af7822 --save-exact
```
`gatsby-plugin-clerk`
```sh
npm i gatsby-plugin-clerk@5.0.0-beta.45 --save-exact
```
`@clerk/localizations`
```sh
npm i @clerk/localizations@2.4.8-snapshot.v2af7822 --save-exact
```
`@clerk/nextjs`
```sh
npm i @clerk/nextjs@5.2.4-snapshot.v2af7822 --save-exact
```
`@clerk/clerk-react`
```sh
npm i @clerk/clerk-react@5.2.8-snapshot.v2af7822 --save-exact
```
`@clerk/remix`
```sh
npm i @clerk/remix@4.2.3-snapshot.v2af7822 --save-exact
```
`@clerk/clerk-sdk-node`
```sh
npm i @clerk/clerk-sdk-node@5.0.16-snapshot.v2af7822 --save-exact
```
`@clerk/shared`
```sh
npm i @clerk/shared@2.3.3-snapshot.v2af7822 --save-exact
```
`@clerk/tanstack-start`
```sh
npm i @clerk/tanstack-start@0.1.7-snapshot.v2af7822 --save-exact
```
`@clerk/testing`
```sh
npm i @clerk/testing@1.1.11-snapshot.v2af7822 --save-exact
```
`@clerk/themes`
```sh
npm i @clerk/themes@2.1.12-snapshot.v2af7822 --save-exact
```
`@clerk/types`
```sh
npm i @clerk/types@4.8.0-snapshot.v2af7822 --save-exact
```

File: .changeset/hungry-clouds-return.md
Comment (anagstef) [inline .changeset/hungry-clouds-return.md:2]: Should we go with a Minor bump as this is a new feature?

**Review by anagstef — APPROVED** [review]
💯

File: .changeset/hungry-clouds-return.md
Comment (octoper) [reply .changeset/hungry-clouds-return.md:2]: Yeah we probably do need minor, good catch @anagstef !

File: packages/clerk-js/src/ui/elements/DevModeNotice.tsx
Comment (alexcarpenter) [inline packages/clerk-js/src/ui/elements/DevModeNotice.tsx:52]: Can/should we add the translation here?

Comment (octoper) [reply packages/clerk-js/src/ui/elements/DevModeNotice.tsx:52]: Good point, we could add it but not sure what's the use case to have this translated as it will only show up in development. 🤔

Comment (alexcarpenter) [reply packages/clerk-js/src/ui/elements/DevModeNotice.tsx:52]: I guess if someone is developing in a diff language. Non-blocker though!

**Comment by PixelJanitor** [comment]
!preview

**Comment by nikosdouvlis** [comment]
!preview

**Comment by nikosdouvlis** [comment]
@PixelJanitor something's up with the GH comments, but here's your preview link: https://javascript-preview-3511.clerkpreview.com/

**Comment by alexcarpenter** [comment]
@nikosdouvlis @PixelJanitor needs to ensure his Clerk membership status is set to public for `!preview` and `!snapshot` comments to function https://github.com/clerk/javascript/blob/main/docs/PUBLISH.md#previewing-prs

**Comment by nikosdouvlis** [comment]
@alexcarpenter that's true, but the comment is missing even though the action I triggered was completed successfully: https://github.com/clerk/javascript/actions/runs/10171336607/job/28132106414

**Comment by octoper** [comment]
@nikosdouvlis @alexcarpenter The comment is only posted once and gets updated when `!preview` deployment is done, also the URL of the preview remains the same

**Comment by octoper** [comment]
!preview

File: .changeset/hungry-clouds-return.md
Comment (anagstef) [inline .changeset/hungry-clouds-return.md:6]:
```suggestion
Add a development mode notice to components and also introduce `popoverBox` descriptor.
```

**Review by PixelJanitor — APPROVED** [review]
Not a code review approval from me (leaving that to an eng), but a design approval based on a deploy preview review. 👍

**Pull request merged**
